components/kyma-environment-broker/internal/provider/aws_provider.go

```diff
@@ -121,54 +124,105 @@ func MultipleZonesForAWSRegion(region string, zonesCount int) []string {
 	return generatedZones
 }
 
-func generateMultipleAWSZones(zoneNames []string) []*gqlschema.AWSZoneInput {
+/*
+*
+generateAWSZones - creates a list of AWSZoneInput objects which contains a proper IP ranges.
+It generates subnets - the subnets in AZ must be inside of the cidr block and non overlapping. example values:
+cidr: 10.250.0.0/16
+  - name: eu-central-1a
+    workers: 10.250.0.0/19
+    public: 10.250.32.0/20
+    internal: 10.250.48.0/20
+  - name: eu-central-1b
+    workers: 10.250.64.0/19
+    public: 10.250.96.0/20
+    internal: 10.250.112.0/20
+  - name: eu-central-1c
+    workers: 10.250.128.0/19
+    public: 10.250.160.0/20
+    internal: 10.250.176.0/20
+*/
+func generateAWSZones(workerCidr string, zoneNames []string) []*gqlschema.AWSZoneInput {
 	var zones []*gqlschema.AWSZoneInput
 
-	// generate subnets - the subnets in AZ must be inside of the cidr block and non overlapping. example values:
-	//vpc:
-	//cidr: 10.250.0.0/16
-	//zones:
-	//	- name: eu-central-1a
-	//workers: 10.250.0.0/19
-	//public: 10.250.32.0/20
-	//internal: 10.250.48.0/20
-	//	- name: eu-central-1b
-	//workers: 10.250.64.0/19
-	//public: 10.250.96.0/20
-	//internal: 10.250.112.0/20
-	//	- name: eu-central-1c
-	//workers: 10.250.128.0/19
-	//public: 10.250.160.0/20
-	//internal: 10.250.176.0/20
-	workerSubnetFmt := "10.250.%d.0/19"
-	lbSubnetFmt := "10.250.%d.0/20"
-	for i, name := range zoneNames {
+	cidr, _ := netip.ParsePrefix(workerCidr)
+	workerPrefixLength := cidr.Bits() + 3
+	workerPrefix, _ := cidr.Addr().Prefix(workerPrefixLength)
+
+	// delta - it is the difference between "public" and "internal" CIDRs, for example:
+	//    WorkerCidr:   "10.250.0.0/19",
+	//    PublicCidr:   "10.250.32.0/20",
+	//    InternalCidr: "10.250.48.0/20",
+	// 4 * delta  - difference between two worker (zone) CIDRs
+	delta := big.NewInt(1)
+	delta.Lsh(delta, uint(31-workerPrefixLength))
+
+	// base - it is an integer, which is based on IP bytes
+	base := new(big.Int).SetBytes(workerPrefix.Addr().AsSlice())
+
+	for _, name := range zoneNames {
```

Comment:
Is number of zones limited somehow?

Reply:
For 6 zones (6 iterations) we will create the zone outside the CIDR from parameter `workerCidr`